spec/ddtrace/contrib/sucker_punch/patcher_spec.rb

```diff
@@ -0,0 +1,101 @@
+require 'ddtrace/contrib/support/spec_helper'
+require 'sucker_punch'
+require 'ddtrace'
+require_relative 'dummy_worker'
+
+RSpec.describe 'Sinatra instrumentation' do
+  before do
+    Datadog.configure do |c|
+      c.use :sucker_punch
+    end
+
+    ::SuckerPunch::Queue.clear
+    ::SuckerPunch::RUNNING.make_true
+  end
+
+  after do
+    ::SuckerPunch::Queue.clear
+  end
+
+  around do |example|
+    # Reset before and after each example; don't allow global state to linger.
+    Datadog.registry[:sucker_punch].reset_configuration!
+    example.run
+    Datadog.registry[:sucker_punch].reset_configuration!
+  end
+
+  context 'successful job' do
+    subject(:dummy_work_success) { ::DummyWorker.perform_async }
+
+    it 'should generate two spans, one for pushing to enqueue and one for the job itself' do
+      is_expected.to be true
+      try_wait_until { fetch_spans.length == 2 }
+      expect(spans.length).to eq(2)
+    end
+
+    it 'should instrument successful job' do
+      is_expected.to be true
+      try_wait_until { fetch_spans.length == 2 }
+
+      span = spans.find { |s| s.resource[/PROCESS/] }
```

Comment:
You might want to extract this statement into a `let` clause:
```ruby
let(:span) { spans.find { |s| s.resource[/PROCESS/] } }
```

Or even (but this might be over-engineering):
```ruby
let(:span) { spans.find { |s| s.resource[resource] } }
let(:resource) { /PROCESS/ }
```

Reply:
makes sense, updated